liota/dccs/iotcc.py

```diff
@@ -267,12 +268,18 @@ def _format_data(self, reg_metric):
             }]
         })
 
-    def set_organization_group_properties(self, reg_entity_name, reg_entity_id, reg_entity_type, entity_local_uuid,
-                                          properties):
-        log.info("Organization Group Properties defined for resource {0}".format(reg_entity_name))
+    def set_system_properties(self, reg_entity_obj, system_properties):
+        if isinstance(reg_entity_obj, RegisteredMetric):
```

Comment:
Why should we set system properties for Metrics?
In create_relationship() which is the only place might call set_system_properties(), for registered metrics, it even does not call this method.
It seems even in the future it does not make sense, e.g., to assign a Organization Group to a metric. Metric must be for certain device or edgesystem, which has Organization groups.

Reply:
We are not setting system property for registered metric, we are setting for its entity. The set_system_proeprties is also called from the user code(check line no 79 of "packages/user_packages/iotcc_mqtt.py" in this PR) so it is better we handle use case for all the entities.

Reply:
Yes, I saw and understood this method could be called by user code. When we create a device, we must create relationship for it, so its system properties could be set. When we create a metric, even user called it, we can just do nothing, since device's system properties must have been set. Why should we set again?

Even if user does not create relationship for that device, when we create metric and call this method, it will only put NULL system properties as input parameters to set to device again. It does nothing useful.

So whether the user code calls this method or not does not matter. As long as a metric tries to call set system properties, we do not need to do anything.

Reply:
The usecase which you have mentioned occurs only when user explicitly makes this call with metric object from user code which we have documented shouldn't be done.
The mechanism of set_properties method is followed  over here because even that sets the properties again for device if the metric object is passed. This usecase is supported as we might have metric as seperate object in future.

Reply:
I think set_properties and set_system_properties are two different things. Of course, we can set properties for metrics, although currently it goes to its parent entity. In fact, both methods finally put k:v pair into property list. Just for our Pulse agent case, some properties you have to fetch from system settings, then set.

But for system properties, in fact, both in your design and in our usage, it is clear that metrics do not need to have system properties. The system properties are more like special workaround for our iotcc and AW. If metrics want to have some properties, we can use set_properties method. Even in the future we can have metrics as separate object, it's more likely we will still use set_properties method.

So, in set_system_properties(), in case of metric object being passed, can just do nothing. I cannot understand  why we should check in some codes which is in fact useless now. 
In the future, if we do have metric as separate object in the future, both set_properties() and set_properties() must to be changed since they will not point to parent any more. Then, at that time, adding codes which will be really used then makes sense.

Reply:
A metric certainly should have k:v pairs associated with it. E.g., units=degreesKelvin, accuracy, precision, etc. Right now we put these in the k:v store of the MO but more properly associated stream of values.